app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -350,29 +352,32 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
 /// Returns true if at least one approved review and no outstanding change
 /// request reviews.
 bool _checkApproval(
+  String authorAssociation,
   List<Map<String, dynamic>> reviewNodes,
   Set<String?> changeRequestAuthors,
 ) {
   assert(changeRequestAuthors.isEmpty);
-  bool hasAtLeastOneApprove = false;
+  final Set<String> allowedReviewers = <String>{'MEMBER', 'OWNER'};
+  final bool authorIsOwnerOrMember = allowedReviewers.contains(authorAssociation);
+  final Set<String?> approvers = <String?>{};
   for (Map<String, dynamic> review in reviewNodes) {
     // Ignore reviews from non-members/owners.
-    if (review['authorAssociation'] != 'MEMBER' && review['authorAssociation'] != 'OWNER') {
+    if (!allowedReviewers.contains(review['authorAssociation'])) {
       continue;
     }
 
     // Reviews come back in order of creation.
     final String? state = review['state'] as String?;
     final String? authorLogin = review['author']['login'] as String?;
     if (state == 'APPROVED') {
-      hasAtLeastOneApprove = true;
+      approvers.add(authorLogin);
       changeRequestAuthors.remove(authorLogin);
     } else if (state == 'CHANGES_REQUESTED') {
       changeRequestAuthors.add(authorLogin);
     }
   }
-
-  return hasAtLeastOneApprove && changeRequestAuthors.isEmpty;
+  final bool approved = (approvers.length > 1) || (authorIsOwnerOrMember && approvers.isNotEmpty);
+  return approved && changeRequestAuthors.isEmpty;
```

Comment:
i think this will have a false positive in the case where the author is a member and self-approves (you'll count the author twice).

I would recommend just adding the author to `approvers` if `authorIsOwnerOrMember`, and then the final `return` can just be `(approvers.length > 1) && changeRequestAuthors.isEmpty`.

Reply:
Self approval was removed at the org level: 
![image](https://user-images.githubusercontent.com/54371434/139929622-327639a3-2f91-4da7-aa7f-0398950c8d65.png)

Added the logic to validate self approval as it doesn't hurt to have there.

Reply:
We should probably add an assertion that this is the case then - it's not entirely obvious to an org manager how changing that setting might impact the bot, right?

Reply:
Actually I applied Ian's suggestion, now the bot covers both scenarios. If self approval is not enabled at the org/repo level and a member/owner is self approving the bot will ignore the approval coming from the PR author.